go-controller/cmd/ovn-k8s-overlay/app/common.go

```diff
@@ -0,0 +1,504 @@
+}
+
+// Create a logical switch for the node and connect it to the distributed router.  This switch will start with one logical port (A OVS internal interface).
+// 1.  This logical port is via which a node can access all other nodes and the containers running inside them using the private IP addresses.
+// 2.  When this port is created on the master node, the K8s daemons become reachable from the containers without any NAT.
+// 3.  The nodes can health-check the pod IP addresses.
+func createManagementPort(nodeName, localSubnet, clusterSubnet string) error {
+	// Create a router port and provide it the first address in the 'local_subnet'.
+	ip, localSubnetNet, err := net.ParseCIDR(localSubnet)
+	if err != nil {
+		return fmt.Errorf("failed parse localsubnet %v : %v", localSubnetNet, err)
+	}
+	ip = util.NextIP(ip)
+	n, _ := localSubnetNet.Mask.Size()
+	routerIPMask := fmt.Sprintf("%s/%d", ip.String(), n)
+	routerIP := ip.String()
+
+	routerMac, stderr, err := util.RunOVNNbctl("--if-exist", "get", "logical_router_port", "rtos-"+nodeName, "mac")
+	if err != nil {
+		logrus.Errorf("Failed to get routerMac,stderr: %q, error: %v", stderr, err)
+		return err
+	}
+
+	var clusterRouter string
+	if routerMac == "" {
+		routerMac = util.GenerateMac()
+		clusterRouter, err = getK8sClusterRouter()
+		if err != nil {
+			return err
+		}
+
+		_, stderr, err = util.RunOVNNbctl("--may-exist", "lrp-add", clusterRouter, "rtos-"+nodeName, routerMac, routerIPMask)
+		if err != nil {
+			logrus.Errorf("Failed to add logical port to router, stderr: %q, error: %v", stderr, err)
+			return err
+		}
+	}
+
+	// Create a logical switch and set its subnet.
+	stdout, stderr, err := util.RunOVNNbctl("--", "--may-exist", "ls-add", nodeName, "--", "set", "logical_switch", nodeName, "other-config:subnet="+localSubnet, "external-ids:gateway_ip="+routerIPMask)
+	if err != nil {
+		logrus.Errorf("Failed to create a logical switch, stdout: %q, stderr: %q, error: %v", stdout, stderr, err)
+		return err
+	}
+
+	// Connect the switch to the router.
+	stdout, stderr, err = util.RunOVNNbctl("--", "--may-exist", "lsp-add", nodeName, "stor-"+nodeName, "--", "set", "logical_switch_port", "stor-"+nodeName, "type=router", "options:router-port=rtos-"+nodeName, "addresses="+"\""+routerMac+"\"")
+	if err != nil {
+		logrus.Errorf("Failed to add logical port to switch, stdout: %q, stderr: %q, error: %v", stdout, stderr, err)
+		return err
+	}
+
+	// Make sure br-int is created
+	stdout, stderr, err = util.RunOVSVsctl("--", "--may-exist", "add-br", "br-int")
+	if err != nil {
+		logrus.Errorf("Failed to create br-int, stdout: %q, stderr: %q, error: %v", stdout, stderr, err)
+		return err
+	}
+
+	// Create a OVS internal interface
+	interfaceName := "k8s-" + (nodeName[:11])
+	stdout, stderr, err = util.RunOVSVsctl("--", "--may-exist", "add-port", "br-int", interfaceName, "--", "set", "interface", interfaceName, "type=internal", "mtu_request=1400", "external-ids:iface-id=k8s-"+nodeName)
```

Comment:
The MTU should be gotten from the config. The python code does so.

Reply:
updated